salt/modules/cmdmod.py

```diff
@@ -569,21 +569,43 @@ def _get_stripped(cmd):
             ret['retcode'] = 1
             return ret
 
+        if output_encoding is not None:
+            log.debug('Decoding output from command %s using %s encoding',
+                      cmd, output_encoding)
+
         try:
-            out = proc.stdout.decode(__salt_system_encoding__)
-        except AttributeError:
+            out = salt.utils.stringutils.to_unicode(
+                proc.stdout,
+                encoding=output_encoding)
+        except TypeError:
+            # stdout is None
             out = ''
         except UnicodeDecodeError:
-            log.error('UnicodeDecodeError while decoding output of cmd {0}'.format(cmd))
-            out = proc.stdout.decode(__salt_system_encoding__, 'replace')
+            out = salt.utils.stringutils.to_unicode(
+                proc.stdout,
+                encoding=output_encoding,
```

Comment:
Will we ever hit this with output_encoding set to None?

Reply:
Yes, `output_encoding` defaults to None so it will always be None unless explicitly passed.